src/network/networkd-fdb.c

```diff
@@ -106,32 +117,23 @@ static int set_fdb_handler(sd_netlink *rtnl, sd_netlink_message *m, Link *link)
 /* send a request to the kernel to add a FDB entry in its static MAC table. */
 int fdb_entry_configure(Link *link, FdbEntry *fdb_entry) {
         _cleanup_(sd_netlink_message_unrefp) sd_netlink_message *req = NULL;
-        sd_netlink *rtnl;
-        Bridge *bridge;
-        uint8_t flags;
         int r;
 
         assert(link);
         assert(link->network);
         assert(link->manager);
         assert(fdb_entry);
 
-        rtnl = link->manager->rtnl;
-        bridge = BRIDGE(link->network->bridge);
-
         /* create new RTM message */
-        r = sd_rtnl_message_new_neigh(rtnl, &req, RTM_NEWNEIGH, link->ifindex, PF_BRIDGE);
+        r = sd_rtnl_message_new_neigh(link->manager->rtnl, &req, RTM_NEWNEIGH, link->ifindex, PF_BRIDGE);
         if (r < 0)
                 return rtnl_log_create_error(r);
 
-        if (bridge)
-                flags = NTF_MASTER;
-        else
-                flags = NTF_SELF;
-
-        r = sd_rtnl_message_neigh_set_flags(req, flags);
-        if (r < 0)
-                return rtnl_log_create_error(r);
+        if (fdb_entry->fdb_ntf_flags != _NEIGHBOR_CACHE_ENTRY_FLAGS_INVALID) {
```

Comment:
Now it is initialized with `_SELF`. And I guess that the flag should be always set here. So, the check is not necessary.

Reply:
yes updated. thanks for review @yuwata .